src/exchange_manager.h

```diff
@@ -95,6 +95,8 @@ class ExchangeManager {
       typename std::set<Bid<T>*>::iterator it4;
       for (it4 = bids.begin(); it4 != bids.end(); ++it4) {
         Bid<T>* b = *it4;
+        Request<T>* r = b->request();
+        double pref = exctx.trader_prefs[b->bidder()][r][b];
```

Comment:
@mbmcgarry can you confirm if changing ``b->bidder()`` to ``r->requester`` fixes the zero pref issue?